formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonDecoder.kt

```diff
@@ -87,6 +87,9 @@ internal open class StreamingJsonDecoder(
             return result
 
         } catch (e: MissingFieldException) {
+            // Add "at path" if and only if we've just caught an exception and it hasn't been augmented yet
+            if (e.message!!.contains("at path")) throw e
+            // NB: we could've use some additional flag marker or augment the stacktrace, but it seemed to be as too much of a burden
             throw MissingFieldException(e.missingFields, e.message + " at path: " + lexer.path.getPath(), e)
```

Comment:
I would prefer `val mfe = .. new mfe with better message ...` to later do `mfe.stackTrace = e.stackTrace`, but unfortunately, it works only on JVM, not sure if we should bother